### PR TITLE
[build] Bump targetSDKVersion and compileSDKVersion to 30, robolectric version to 4.6.1.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Bump targetSDKVersion and compileSDKVersion to 30.</changelog>`.

### Summary of changes

This PR bumps targetSDKVersion and compileSDKVersion to 30.
And also removes the unused `WRITE_EXTERNAL_STORAGE` and `READ_EXTERNAL_STORAGE` from the app module.

### CODE REVIEW

**Review by tobrun — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,8 +4,6 @@
     package="com.mapbox.maps.testapp">
 
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```
Comment (kiryldz) [inline app/src/main/AndroidManifest.xml:8]: Finally! 👍

Comment (pengdev) [reply app/src/main/AndroidManifest.xml:8]: Actually we need it for `com.squareup.leakcanary:leakcanary-android-core:2.4` checking if we could bump the dependency to fix the storage permission.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,8 +4,6 @@
     package="com.mapbox.maps.testapp">
 
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
Comment (kiryldz) [inline app/src/main/AndroidManifest.xml:6]: If we're talking about Android 30 - are we OK with just this permission? afair there were some changes but perhaps it's more about foreground / background location services.
But worth to read more I believe..

Comment (pengdev) [reply app/src/main/AndroidManifest.xml:6]: yes, it definitely worth more investigation, and that's why I converted this PR to draft 😄

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (pengdev) [reply app/src/main/AndroidManifest.xml:8]: ok, by default leakcanary doesn't try to request this permission, so I think it's fine to remove.

**Comment by amelo1990** [comment]
8225715eae95ad17b9221c6fdffb4a3f859511f2

File: plugin-animation/src/test/resources/robolectric.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+sdk=28
```
Comment (pengdev) [inline plugin-animation/src/test/resources/robolectric.properties:1]: Refs https://github.com/mapbox/mapbox-navigation-android/pull/3230#issuecomment-649081774

**Review by kiryldz — APPROVED** [review]

**Pull request merged**